google/cloud/storage/internal/retry_resumable_upload_session.cc

```diff
@@ -57,112 +57,164 @@ StatusOr<ResumableUploadResponse> RetryResumableUploadSession::UploadFinalChunk(
       });
 }
 
+// Implements the retry loop for a resumable upload session.
+//
+// A description of resumable uploads can be found at:
+//     https://cloud.google.com/storage/docs/performing-resumable-uploads
+//
+// A description of the gRPC analog can be found in the proto file:
+//    https://github.com/googleapis/googleapis/blob/master/google/storage/v2/storage.proto
+//
+// Particularly the documentation for `WriteObject()`, `WriteObjectRequest`,
+// `StartResumablewrite()` and `QueryResumableWrite()`.
```

Comment:
Perhaps this would be better before the colon in the previous block.

Reply:
Done. And thanks.